metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_lineage.py

```diff
@@ -162,19 +165,38 @@ def __init__(self, config: SnowflakeV2Config, report: SnowflakeV2Report) -> None
         self.report = report
         self.logger = logger
 
-    # Rewrite implementation for readability, efficiency and extensibility
-    def _get_upstream_lineage_info(
-        self, dataset_name: str
-    ) -> Optional[Tuple[UpstreamLineage, Dict[str, str]]]:
+    def get_workunits(
+        self, discovered_tables: List[str], discovered_views: List[str]
+    ) -> Iterable[MetadataWorkUnit]:
+
+        try:
+            conn = self.get_connection()
+        except Exception as e:
+            if isinstance(e, SnowflakePermissionError):
+                self.report_error("permission-error", str(e))
+            else:
+                logger.debug(e, exc_info=e)
+                self.report_error(
+                    "snowflake-connection",
+                    f"Failed to connect to snowflake instance due to error {e}.",
+                )
+            return
 
-        if self._lineage_map is None or self._external_lineage_map is None:
-            conn = self.config.get_connection()
         if self._lineage_map is None:
-            with PerfTimer() as timer:
-                self._populate_lineage(conn)
-                self.report.table_lineage_query_secs = timer.elapsed_seconds()
+            self._lineage_map = defaultdict(SnowflakeTableLineage)
+            if self.report.edition == SnowflakeEdition.STANDARD:
+                logger.info(
+                    "Snowflake Account is Standard Edition. Table to Table Lineage Feature is not supported."
+                )
```

Comment:
if we do not have enterprise edition, should we continue at all? why not return here?

Reply:
reason mentioned above. I'll also add a comment in snowflake_lineage code at the start.